docs/dx.rst

```diff
@@ -0,0 +1,246 @@
+They are only handled through metadata on the server and as a result, do not
+have a canonical ID of their own. As a result, paths to folders can only be
+`DXVirtualPath`. Accessing the `DXVirtualPath.canonical_path` property for folders
+would raise an error.
+
+The summary of different behaviors for different filesystems is presented here. The
+individual details are explained further below.
+
+.. list-table:: Filesystem Differences
+   :widths: 15 30 25 30
+   :header-rows: 1
+
+   * - Topic
+     - Posix
+     - Swift/S3
+     - DNAnexus
+   * - File/ Directory names
+     - Filenames without extension and dirnames with ext are allowed.
+       A dir and a file can have the same name in same path
+     - Anonymous file / dir names are not allowed on swift paths
+     - Filenames without extension and dirnames with ext are allowed.
+       A dir and a file can have the same name in same path
+   * - Duplicates
+     - Not allowed
+     - Not allowed
+     - Duplicate filenames are allowed. Duplicate dir names are not allowed
+   * - list
+     - Recurisvely lists the files and empty directories
+     - Recurisvely lists the files and empty directory markers.
+     - Recursively lists the files within DX folder path. No empty
+       directories are listed.
+   * - list with prefix
+     - Treated as prefix to absolute path
+     - Treated as prefix to absolute path
+     - Treated as path to a subfolder to list
+   * - list on filepath
+     - Returns [filepath]
+     - Returns [filepath]
+     - Returns []
+   * - copy, target exists and is file
+     - Overwrites the file
+     - Overwrites the file
+     - Deletes the existing file before copying over
+   * - copy, target exists and is dir
+     - Not allowed
+     - Not possible
+     - Copied within existing dir
+   * - serverside copy/copytree
+     - Allowed
+     - Not allowed
+     - Allowed
+   * - copytree, target exists as file
+     - Not allowed
+     - Allowed
+     - Allowed
+   * - copytree, target exists as dir
+     - Merges the two directories
+     - Overwrites the dir
+     - Copies inside existing directory.
+       If root folder is moved, project name is used while copying if needed.
+
+Files on DNAnexus
+-----------------
+Files stored on the DX platform are immutable. This is because the files are
+internally stored in AWS while the metadata handling is taken care of by the
+platform. Hence, once a file is uploaded, it cannot be modified.
+
+When one file is copied to another project, only additional metadata is produced,
+while the underlying file on AWS remains the same. This is essential. The same
+file with the same canonical ID will appear in both projects, and can have
+different folder paths. Deleting a file from one project is possible, which
+deletes the metadata and leaves the file untouched in other projects.
+
+DXPath on stor
+--------------
+All `DXPath` instances on stor must have a project. The project is the first part
```

Comment:
Simpler:

Project is always required for DNAnexus instances:

```
>>> Path('dx://path/to/file')
Traceback (most recent call last):
...
<exception>
```

but projects are normalized.

Reply:
Changed.